skimage/future/graph/graph_cut.py

```diff
@@ -261,21 +340,20 @@ def _ncut_relabel(rag, thresh, num_cuts):
     d, w = _ncut.DW_matrices(rag)
     m = w.shape[0]
 
+    # If 2 regions not cutting is optimal choice
     if m > 2:
         d2 = d.copy()
         # Since d is diagonal, we can directly operate on its data
         # the inverse of the square root
         d2.data = np.reciprocal(np.sqrt(d2.data, out=d2.data), out=d2.data)
 
-        # Refer Shi & Malik 2001, Equation 7, Page 891
+        # Refer Shi & Malik 2000, Equation 7, Page 891
         vals, vectors = linalg.eigsh(d2 * (d - w) * d2, which='SM',
-                                     k=min(100, m - 2))
+                                     k=min(100, m - 1))
 
         # Pick second smallest eigenvector.
-        # Refer Shi & Malik 2001, Section 3.2.3, Page 893
-        vals, vectors = np.real(vals), np.real(vectors)
-        index2 = _ncut_cy.argmin2(vals)
-        ev = vectors[:, index2]
+        # Refer Shi & Malik 2000, Section 3.2.3, Page 893
+        ev = vectors[:, 1]
```

Comment:
Eigenpairs are returned in ascending order, so eigenvector corresponding to second smallest eigenvalue will always be at index 1.

See line 35 here: https://github.com/hpfem/arpack/blob/master/ARPACK/SRC/dseupd.f